src/refiners/training_utils/trainer.py

```diff
@@ -345,12 +349,25 @@ def backward(self) -> None:
         """Backward pass on the loss."""
         self._call_callbacks(event_name="on_backward_begin")
         scaled_loss = self.loss / self.clock.num_step_per_iteration
+        if self.scaler is not None:
+            self.scaler.scale(scaled_loss).backward()  # type: ignore
+        else:
+            backward(tensors=scaled_loss)
         backward(tensors=scaled_loss)
```

Comment:
Don't you have a double backward here?

Reply:
@limiteinductive oh yup lol let me fix that in a sec

Reply:
Fixed i think